operators/src/adapters/raster_time_substream.rs

```diff
@@ -0,0 +1,266 @@
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use crate::util::Result;
+use futures::{
+    ready,
+    stream::{FusedStream, Stream},
+    Future,
+};
+use geoengine_datatypes::{
+    primitives::TimeInterval,
+    raster::{Pixel, RasterTile2D},
+};
+use pin_project::pin_project;
+
+#[must_use = "streams do nothing unless polled"]
+#[pin_project(project = RasterTimeSubStreamAdapterProjection)]
+pub struct RasterTimeMultiFold<St, P, A, AInit, F, Fut>
+where
+    St: Stream<Item = Result<RasterTile2D<P>>>,
+    P: Pixel,
+    AInit: FnMut() -> A,
+    F: FnMut(A, St::Item) -> Fut,
+    Fut: Future<Output = A>,
+{
+    #[pin]
+    stream: St,
+
+    accumulator_initializer: AInit,
+    f: F,
```

Comment:
please give f a more decriptive name. It's the fn executed for all tiles in the fold right? So m,aybe: fold_fn ?

Reply:
a6cc28f